src/org/opensolaris/opengrok/web/api/v1/controller/SystemController.java

```diff
@@ -35,11 +37,15 @@
 
     private final RuntimeEnvironment env = RuntimeEnvironment.getInstance();
 
+    @Inject
+    private SuggesterService suggester;
+
     @PUT
     @Path("/refresh")
     @Consumes(MediaType.TEXT_PLAIN)
     public void refresh(final String project) {
         env.maybeRefreshIndexSearchers(Collections.singleton(project));
+        suggester.refresh(project);
```

Comment:
I wonder if this is necessary. The refresh here is meant to make the `indexSearcher`s to move to the new index generation. The suggester refresh happens already when project is marked as indexed, no ?

Reply:
Well, from the actual implementation in the `Indexer` this is not necessary. This is only called for partial reindex and only after the `markIndexed` call. Therefore, it will probably rebuild the same data structure for the second time. However, the `markIndexed` call also refreshes index searchers so what is the point of calling refresh again via this method? Or did I get it wrong?

From the logical point of view. This is called to notify the webapp that an index changed; therefore, suggester should react, right?

Reply:
It seems that the refresh done in `main()` via `/refresh` is largely redundant given that the `searcherManager`s are refreshed via `/indexed`. This is not a problem of the suggester, rather than the indexer itself. Filed #2218 to track this.